File: frontend/lib/models/palabra_clave.dart
class PalabraClave {
  final int id;
  final String palabra;
  final String? descripcion;
  final bool activo;
  final int numeroDocumentos;

  PalabraClave({
    required this.id,
    required this.palabra,
    this.descripcion,
    required this.activo,
    this.numeroDocumentos = 0,
  });

  factory PalabraClave.fromJson(Map<String, dynamic> json) {
    return PalabraClave(
      id: json['id'],
      palabra: json['palabra'],
      descripcion: json['descripcion'],
      activo: json['activo'],
      numeroDocumentos: json['numeroDocumentos'] ?? 0,
    );
  }
}

class CreatePalabraClaveDTO {
  final String palabra;
  final String? descripcion;

  CreatePalabraClaveDTO({
    required this.palabra,
    this.descripcion,
  });

  Map<String, dynamic> toJson() {
    return {
      'palabra': palabra,
      'descripcion': descripcion,
    };
  }
}

File: backend/Models/PalabraClave.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SistemaGestionDocumental.Models;

[Table("palabras_clave")]
public class PalabraClave
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [Column("palabra")]
    [StringLength(50)]
    public string Palabra { get; set; } = string.Empty;

    [Column("descripcion")]
    [StringLength(200)]
    public string? Descripcion { get; set; }

    [Column("activo")]
    public bool Activo { get; set; } = true;

    // Relaciones
    public virtual ICollection<DocumentoPalabraClave> DocumentoPalabrasClaves { get; set; } = new List<DocumentoPalabraClave>();
}
